src/views/studio/EditVideoSheet/EditVideoSheet.tsx

```diff
@@ -60,36 +117,195 @@ export const EditVideoSheet: React.FC = () => {
+      return
+    }
+
+    if (queryNodeState.indexerHead >= transactionBlock) {
+      setTransactionStatus(ExtrinsicStatus.Completed)
+      transactionCallback?.()
+    }
+  }, [queryNodeState, transactionBlock, transactionCallback, transactionStatus])
+
+  const handleSubmit = createSubmitHandler(async (data) => {
+    if (!video.url || !video.blob) {
+      setFileSelectError('Video was not provided')
+      return
+    }
+    if (!thumbnail.url || !thumbnail.blob) {
+      setFileSelectError('Thumbnail was not provided')
+      return
+    }
+    if (!joystream || !memberId || !channelId) {
+      return
+    }
+
+    setFileSelectError(null)
+
+    setTransactionStatus(ExtrinsicStatus.ProcessingAssets)
+
+    const metadata: CreateVideoMetadata = {
+      ...(data.title ? { title: data.title } : {}),
+      ...(data.description ? { description: data.description } : {}),
+      ...(data.category ? { category: Number(data.category) } : {}),
+      ...(data.isPublic != null ? { isPublic: data.isPublic } : {}),
+      ...(data.hasMarketing != null ? { hasMarketing: data.hasMarketing } : {}),
+      ...(data.isExplicit != null ? { isExplicit: data.isExplicit } : {}),
+      ...(data.language ? { language: data.language } : {}),
+      ...(data.publishedBeforeJoystream
+        ? {
+            publishedBeforeJoystream: {
+              isPublished: true,
+              date: data.publishedBeforeJoystream.toString(),
+            },
+          }
+        : {}),
+      ...(video.mimeType
+        ? {
+            mediaType: {
+              mimeMediaType: video.mimeType,
+            },
+          }
+        : {}),
+      ...(video.duration ? { duration: Math.round(video.duration) } : {}),
+      ...(video.mediaPixelHeight ? { mediaPixelHeight: video.mediaPixelHeight } : {}),
+      ...(video.mediaPixelWidth ? { mediaPixelWidth: video.mediaPixelWidth } : {}),
+    }
+
+    const assets: VideoAssets = {}
+    let videoContentId = ''
+    let thumbnailContentId = ''
+
+    if (video.blob && videoHashPromise) {
+      const [asset, contentId] = joystream.createFileAsset({
+        size: video.blob.size,
+        ipfsContentId: await videoHashPromise,
+      })
+      assets.video = asset
+      videoContentId = contentId
+    } else {
+      console.warn('Missing video data')
+    }
+
+    if (thumbnail.blob && thumbnailHashPromise) {
+      const [asset, contentId] = joystream.createFileAsset({
+        size: thumbnail.blob.size,
+        ipfsContentId: await thumbnailHashPromise,
+      })
+      assets.thumbnail = asset
+      thumbnailContentId = contentId
+    } else {
+      console.warn('Missing thumbnail data')
+    }
+    let assetsOwner: VideoId = ''
+
+    try {
+      const { data: newVideoId, block } = await joystream.createVideo(
+        memberId,
+        channelId,
+        metadata,
+        assets,
+        (status) => {
+          setTransactionStatus(status)
+        }
+      )
+      assetsOwner = newVideoId
```

Comment:
Channel is the owner of the assets, this is used to filter only assets for a given channel in the uploads view